### ISSUE TITLE
"Wrong" treeview-links to elements with only brief description

### ISSUE THREAD

**Issue by thom-sch** [doxygen/doxygen#11259]
First I thought this is a bug, but I've figured out that this is (incorrect?) behavior of the browsers - observed on Firefox, Edge, Chrome and Opera, others not tested.

**Describe the bug**
I see:
Having a long list of define's, some of them with only brief descripton, others with detailed description.
Clicking on the treeview link to navigate to the respective description, navigation is done properly only for define's with detailed description.
Navigation for defines with only brief description depends on where the cursor actually is located:
- above the link destination (i.e. above the linked define) navigation is partly correct
- below the link destination: the window is not properly scrolled to the desired destination

I've figured out:
Doxygen creates two references for each define:
- r_[...verylongnumber..] which links to the brief description
- [...verylongnumber...] which links to the detailed description
In case there is no detailed description doxygen places the "detailed" anchor inside the `<td>-tag` of the brief description, to cause links to the (not present) detailed description are routed to the brief description.
But the location inside the `<td>-tag` seams to confuse the browsers: the window isnot properly scrolled: if the cursor is located more down the window scrolls more and more wrong.
Placing the "detailed" anchor just **before** the `<td>-tag` or the `<tr>-tag` all works fine!

**To Reproduce**
See attached Project:
In file **`_modul1_8cpp.html`** I've **manually** moved the "detailed" anchor of Macro1 from inside the `<td>-tag` to outside the `<td>-tag` - and all is fine for Macro1 (you may also move it before the `<tr>-tag`).
You can see the behavior doing:
- with the treeview navigate to "Modu1.cpp" and expand it
- then click on "Macro10": the window scrolls properly to "Macro10"
- then click on "Macro2" or Macro5..9 (Macros with only brief description): The window does not scroll properly
- Click on "Macro1" (also with only brief description but with the anchor before the `<td>-tag`): The window properly scrolls to "Macro1"

As a workaround you can set `ALWAYS_DETAILED_SEC = YES`

**Additionally**
This behavior is not restricted to defines but also belongs to other elements, i.e. variables etc.

**Version**
Actual Version 1.12.0 on Windows10, but I think this is a (incorrect?) browser behavior and so its not dependent on the doxygen version.

[Test23.zip](https://github.com/user-attachments/files/18035223/Test23.zip)

regards from Meerbusch
Thomas

**Comment by albert-github** [comment]
Unfortunately there is no settings file present, so I used:
```
QUIET = YES
GENERATE_TREEVIEW = YES
INPUT = src
```

With the doxygen 1.12.0 version I can reproduce the problem. with the current master version (1.13.0 (8007b6cfff3fdbafb38c15b484afeac12b2bb812)) it looks to me that all is OK (tested on Windows with firefox and chrome).

I think the problem is fixed but not released, most likely by means of:
```
Commit: 0fab7bcce3655fd4dda09ab783cbc682c94ec90a [0fab7bc]
Date: Monday, September 9, 2024 9:27:16 PM

More fixes for strong enum values

- When two strong enums at the same scope level had an enum value with
  the same name, the wrong one could be linked to from the source browser and inline code.
- The naviation tree showed bogus non-linkable values for namespace members that were
  listed under a file entry.
- Scrolling to an enum value from the navigation tree caused the page to keep scrolling
```
especially the fix in `navtree.js`

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.13.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
